### PR TITLE
Persistent Map Voting

### PR DESCRIPTION
**Pull request by adacovsk**
[system]

Votes remain until map is picked, then that maps votes are reset. I don't know what this voting method is called, but according to Jeff, Colonel Marines uses it and it seems effective. Popular maps will continue to be played almost exclusively, but once lesser popular maps get enough votes for after many rounds, they can win the vote. This prevents:
- angry people when weighted random voting picks maps with low percent chance
- angry people when majority vote picks almost exclusively being box
- angry people when their vote means nothing

This method  also incentivizes for playing the round. With dedication or autism, they can eventually sway the map pick by themselves. This voting method also preserves all the conditions for maps (>25 players, <15 players, etc.) too. 

Also, I'm not removing the other voting methods, just including this one.

tl;dr:
- Keep votes throughout rounds
- Only next map gets votes reset

:cl:
 * rscadd: Added new map voting method available: persistent votes through rounds! Weighted is still being used.
 * tweak: rewrote some voting code

### CODE REVIEW

**Comment by Anton-Kr** [comment]
I dont like it

**Comment by gurfan** [comment]
Interesting idea, one suggestion I want to add is that votes shouldn't persist after the voter has left the game.

**Comment by SonixApache** [comment]
maybe if i'm not lazy i can adapt this for the auto-voteban system
my complaint remains that this would cause meme maps to be forced through every two to three rounds rather than at the frequency they appear at now

**Comment by adacovsk** [comment]
> maybe if i'm not lazy i can adapt this for the auto-voteban system my complaint remains that this would cause meme maps to be forced through every two to three rounds rather than at the frequency they appear at now

Well, with weighted, you typically need 3 votes to get your map selected with low chances (25% chance). You're going to need at 3-4 rounds if you assume box gets like 12 votes each time if you assume the same voting pattern. The likelihood is about the same.

I'm probably going to have to mess with voting.dm and make it more usable

**Comment by gurfan** [comment]
> maybe if i'm not lazy i can adapt this for the auto-voteban system my complaint remains that this would cause meme maps to be forced through every two to three rounds rather than at the frequency they appear at now

The obvious solution is to remove meme maps

**Comment by Eneocho** [comment]
Lets see where this goes.
Prepare for unforeseen consequences.

**Comment by Exxion** [comment]
something something remove some small number of votes per round even if the map wasn't picked

**Comment by Kromkar** [comment]
What is with all these changes to the tried and true weighted voting system? Truly coders never play so why do you even care

**Comment by wotisjanitor** [comment]
cringe.

**Comment by GreatBigFailure** [comment]
I see nothing wrong with this. I like it. I guess it could be a problem if one person is able to eventually force weighted to roll snaxi every day but that seems like a small price. Worth a try

**Comment by GreatBigFailure** [comment]
Seems like I'm of a losing opinion on this one.
Exxion is probably right, either remove a static amount of votes per round, or some malleable amount based on either weighted percentages or population or something. I get the appeal of having one person eventually be able to force their favorite map by just playing enough rounds, my favorites are probably synergy and waystation so I really like that element, but it'll also be used to force snow and what not even more. 
Maybe a maximum amount of persistent votes? Like keep it at just enough to make rolling the map possible. Just spit balling here because I think this is a good idea, but I'm alone in that.

**Comment by adacovsk** [comment]
> Seems like I'm of a losing opinion on this one. Exxion is probably right

I'll work on what Exxion has said. It would be simple to do after I completely rework this horrible code in voting.dm. Just like 1 vote off each map per round would be very simple

**Comment by Viciousaurus** [comment]
I'll be angry no matter what.
